www/src/components/CodeBlock.tsx

```diff
@@ -32,12 +32,30 @@ export type CollapsibleLiveEditorTypes = {
 
 function CollapsibleLiveEditor({ children }: CollapsibleLiveEditorTypes) {
   const [collapseIsOpen, setCollapseIsOpen] = useState(false);
+
+  const eventSegmentSubmitter = (e: Event & { target: HTMLElement }) => {
+    const useCaseElement: { parentNode: any } = e.target.closest('.pgn-doc__code-block') as HTMLElement;
+    let elementBeforeUseCase: {
+      className: string, id: string, previousSibling: any
+    } = useCaseElement.parentNode.previousSibling;
+    const componentNameAndCategory: string = window.location.pathname.replace(/\//g, '.');
+
+    while (elementBeforeUseCase.className !== 'pgn-doc__heading') {
+      elementBeforeUseCase = elementBeforeUseCase.previousSibling;
+    }
+
+    global.analytics.track(`openedx.ui${componentNameAndCategory}${elementBeforeUseCase.id}.${collapseIsOpen ? 'close' : 'open'}`, {
+      value: `${componentNameAndCategory.replace(/.components./gi, '')}${elementBeforeUseCase.id}`,
+    });
+  };
+
   return (
     <div className="pgn-doc__collapsible-live-editor">
       <Collapsible.Advanced
         unmountOnExit={false}
         open={collapseIsOpen}
         onToggle={(isOpen: boolean) => setCollapseIsOpen(isOpen)}
+        onClick={(e) => eventSegmentSubmitter(e)}
```

Comment:
nit: it's not necessary to use arrow function in this case, you could just do `onClick={eventSegmentSubmitter}`

Reply:
Yes, thanks